### PR TITLE
Make listCmd honor hint:cc:off

### PR DESCRIPTION
**Pull request by Clyybber**
Amends https://github.com/nim-lang/Nim/pull/13603

### CODE REVIEW

**Comment by Clyybber** [comment]
Eh, there is value to showing when the link step begins, as it can take quite a while (especially with lto).

**Comment by timotheecour** [comment]
I wish https://github.com/nim-lang/Nim/pull/13516 (sent out 12 days ago) could get merged first since this will cause conflicts ... @araq is there anything else needed on https://github.com/nim-lang/Nim/pull/13516 ? (the only comment  https://github.com/nim-lang/Nim/pull/13516#issuecomment-592438946 was about build_all.sh, not something actionable for that PR)

File: compiler/extccomp.nim [hunk 1]
```diff
@@ -704,22 +704,6 @@ proc addExternalFileToCompile*(conf: ConfigRef; filename: AbsoluteFile) =
     flags: {CfileFlag.External})
   addExternalFileToCompile(conf, c)
 
-proc compileCFiles(conf: ConfigRef; list: CfileList, script: var Rope, cmds: var TStringSeq,
```
Comment (Araq) [inline compiler/extccomp.nim:713]: why inline this proc?

Comment (Clyybber) [reply compiler/extccomp.nim:713]: It's only used once and I found it confusing/misleading since it doesnt actually compile the C files.

**Comment by timotheecour** [comment]
@Clyybber needs rebase now that https://github.com/nim-lang/Nim/pull/13516 got merged; let me know if you have any questions since there is some overlap in logic

**Comment by Clyybber** [comment]
Note (due to request): This also fixes an obscure HCR bug :)

**Pull request merged**